src/Codec/Archive/Zip.hs

```diff
@@ -351,7 +352,7 @@ checkPath fp =
   maybe (E.throwIO (UnsafePath fp)) (\_ -> return ())
     (resolve . splitDirectories $ fp)
   where
-    resolve :: Monad m => [String] -> m [String]
+    resolve :: (MonadFail m, Monad m) => [String] -> m [String]
```

Comment:
I've dealt with this another way (just removing the type sig and letting the compiler infer).